crates/router/src/core/payments/helpers.rs

```diff
@@ -3043,6 +3046,69 @@ pub async fn make_ephemeral_key(
     Ok(services::ApplicationResponse::Json(ek))
 }
 
+#[cfg(feature = "v2")]
+pub async fn make_ephemeral_key(
+    state: SessionState,
+    customer_id: id_type::GlobalCustomerId,
+    merchant_id: id_type::MerchantId,
+    headers: &actix_web::http::header::HeaderMap,
+) -> errors::RouterResponse<EphemeralKeyResponse> {
+    let store = &state.store;
+    let id = id_type::EphemeralKeyId::generate_key_id("eki");
+    let secret = format!("epk_{}", &Uuid::new_v4().simple().to_string());
+    let resource_type = services::authentication::get_header_value_by_key(
+        headers::X_RESOURCE_TYPE.to_string(),
+        headers,
+    )?
+    .map(ephemeral_key::ResourceType::from_str)
+    .transpose()
+    .change_context(errors::ApiErrorResponse::InvalidRequestData {
+        message: format!("`{}` header is invalid", headers::X_RESOURCE_TYPE),
+    })?
+    .get_required_value("ResourceType")
```

Comment:
is this always tied to a customer? we might create the ephemeral key for payment as well right?